docs/storage/upload/upload.go

```diff
@@ -0,0 +1,142 @@
+// Copyright 2017 Google Inc. All rights reserved.
+// Use of this source code is governed by the Apache 2.0
+// license that can be found in the LICENSE file.
+
+package main
+
+import (
+	"flag"
+	"fmt"
+	"io"
+	"log"
+	"os"
+	"path/filepath"
+
+	"google.golang.org/api/storage/v1"
+
+	"golang.org/x/net/context"
+	"golang.org/x/oauth2/google"
+)
+
+func main() {
+	// Prevent log from printing out time information
+	log.SetFlags(0)
+
+	var projectID, bucket, source, name string
+	var public bool
+
+	flag.StringVar(&bucket, "bucket", "test", "the bucket to upload content to")
+	flag.StringVar(&projectID, "project-id", "golang-samples", "the ID of the GCP project to use")
```

Comment:
A project ID and bucket default is not useful. What about falling back to a env variable value for both if they are not set?